cmd/rosa/main.go

```diff
@@ -105,3 +113,27 @@ func main() {
 		os.Exit(1)
 	}
 }
+
+func versionCheck() {
+	rprtr := reporter.CreateReporterOrExit()
+
+	currVersion, err := gversion.NewVersion(info.Version)
+	if err != nil {
+		_ = rprtr.Errorf("There was a problem verifying the current version of rosa. Please be aware that you are possibly running an outdated version: %w", err)
+		os.Exit(1)
+	}
+
+	latestVersionFromMirror, err := rosa.RetrieveLatestVersionFromMirror()
+	if err != nil {
+		_ = rprtr.Errorf("There was a problem retrieving latest version from mirror. Please be aware that you are possibly running an outdated version: %w", err)
+		os.Exit(1)
+	}
+
+	if !currVersion.Equal(latestVersionFromMirror) {
```

Comment:
I went with an equality comparison here, but I could also see a use case for _only_ checking `currVersion.LessThan` like in /cmd/verify/rosa/cmd.go#L61 . One pain point I could see is that comparing only for equality could lead to a prompt every time someone runs a command from a development build.

In osdctl, we solved this problem by adding a `-S` flag to skip the version check altogether - I could go either way. Let me know if you'd like me to handle this differently.

Reply:
I think I would lean toward `LessThan` to avoid noise from dev builds. As this will be the case for roughly 2 weeks every sprint during the release process.